### PR TITLE
[OneWire] bug fixes and new multi-sensors

### LINKED ISSUE #6434: [OneWire] Missing iButtonLink MS-TL, MS-TPD, and MS-TC
The OneWire binding is missing support for several of iButtonLink's multisensor products, including the MS-TL (temp & luminance), MS-TC (temp & current), and MS-TPD (temp & air flow). The only one of the three I actually own is the MS-TL, but it'd be nice if all are supported.

### LINKED ISSUE #6750: [onewire] slave missing error for temperature thing
@J-N-K I use your last jar from https://github.com/eclipse/smarthome/issues/6728#issuecomment-450370556

I get:
>23:06:12.846 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - 'onewire:temperature:server1:Ow_T_KG_DG_VL' changed from OFFLINE (COMMUNICATION_ERROR): slave missing to ONLINE

but this thing doesn't has slaves
> Thing temperature Ow_T_KG_DG_VL [ id="28.A26385050000", refresh=60 ] // Sensor 259

### LINKED ISSUE #6751: [onewire] BMS-THS-211 not working as BMS
@J-N-K I have BMS-THS-211 from Elab.
If I discover them it get BMS sensors at inbox. I didn't add them.
If I configure them manually I get `UNINITIALIZED - HANDLER_CONFIGURATION_PENDING`
If I configure them as MS-TH they will work.
Which HIH is the default if nothing is configured?

This multi sensor has an HIH-4031 which should be compatible with HIH-4000.
Can you give a note to this (BMS-THS and HIH-4031) in your bindings docu?

I think they give you the right values for an auto config.
Here are some more infos:
```
# owget /26.22AE7F010000/pages/page.ALL | hexdump 
0000000 d00f 0106 0702 0000 93dd 0004 10ff fc00
0000010 0000 0000 0000 0000 0019 0000 0000 0000
0000020 0000 0000 0000 0000 070c 15dd 0000 0000
0000030 d628 5aec 0004 0000 0000 0000 0b09 0000
0000040

# owget /26.22AE7F010000/MultiSensor/type
MS-TH

```

### LINKED ISSUE #6759: [onewire] smartmeter binding is blocked while onewire handles missing sensors
I use OH2.4.0.
I have seen this first with binding from OH2.4.0 and OH2.5.0 snapshot.
Now I see this with the last build from @J-N-K [jar](https://janessa.me/esh/org.eclipse.smarthome.binding.onewire-0.11.0-SNAPSHOT-da.jar).

I don't know if KNX or other bindings are blocked too but smartmeter is the binding with the highest frequency and showed connection timeouts https://github.com/openhab/openhab2-addons/issues/4462.
The frequency from smartmeter drops from 2 sec to more than 90 secs per value. 

This happens if some sensors are missing because one of my owservers are going down.
The old onewire binding from OH2.3 never had this problem.

While this error the CPU load of OH goes down to nearly zero.

I hope everybody is aware that a binding can affect other binding in such a dimension.

### PR DESCRIPTION
**Pull request by J-N-K**
Fixes #6434
Fixes #6750
Fixes #6751
Fixes #6759 

- bug fixes for existing multisensors
- adds new multisensors
- bug fixes for channel configuration in textual defined channels
- bug fix for improper intialization of bridges after edits
- updates to the documentation
- fixes blocking in case of many failing sensor property updates

### CODE REVIEW

**Comment by J-N-K** [comment]
What is the proper way to deprecate thing-types? My problem is: Thex are still fully functional, it just needs additional code (which I would like to remove). I can upgrade the thing-type automatically for PaperUI defined things.

**Comment by maggu2810** [comment]
Hm, if "migrate thing type" does not work for non managed providers and the change is necessary, the product needs to convert that file (on disk or on demand) or needs to provide a big fat warning / upgrade instructions.

**Comment by J-N-K** [comment]
"migrate thing-type" works "on-the-fly", that is during runtime as long as the thing-type is still present (that's what I do at the moment: if it is an old thing-type, it is upgraded to the new thing type in `initialize`). 

I feel that we should look for a framework solution for migrating thing-types for non-managed providers.

File: extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/BasicMultisensorThingHandler.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                DS2438Configuration ds2438configuration = new DS2438Configuration(pages);
+
+                sensorType = ds2438configuration.getSensorSubType();
+                properties.put(PROPERTY_MODELID, sensorType.toString());
+
+                String vendor = ds2438configuration.getVendor();
+                properties.put(PROPERTY_VENDOR, vendor);
+            }
+        } catch (OwException e) {
+            logger.info("updating thing properties failed: {}", e.getMessage());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/BasicMultisensorThingHandler.java]: reduce to debug

File: extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/BasicMultisensorThingHandler.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            return;
+        }
+
+        validConfig = true;
+        updatePresenceStatus(UnDefType.UNDEF);
+    }
+
+    @Override
+    protected void updateSensorProperties() {
+        logger.info("updating sensor properties of {}", thing.getLabel());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/BasicMultisensorThingHandler.java]: reduce to debug

File: extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/AdvancedMultisensorThingHandler.java [hunk 3]
```diff
@@ -281,7 +283,7 @@ private void configureThingChannels() {
 
     @Override
     protected void updateSensorProperties() {
-        logger.info("updating {}", thing.getLabel());
+        logger.info("updating sensor properties of {}", thing.getLabel());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/AdvancedMultisensorThingHandler.java]: reduce to debug

File: extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/AdvancedMultisensorThingHandler.java [hunk 4]
```diff
@@ -298,12 +300,20 @@ protected void updateSensorProperties() {
...
+            properties.put(PROPERTY_MODELID, sensorType.toString());
+
+            String vendor = ds2438configuration.getVendor();
+            properties.put(PROPERTY_VENDOR, vendor);
+
+            properties.put(PROPERTY_PROD_DATE, ds2438configuration.getProductionDate());
+            properties.put(PROPERTY_HW_REVISION, ds2438configuration.getHardwareRevision());
+
         } catch (OwException e) {
             logger.info("updating thing properties failed: {}", e.getMessage());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/AdvancedMultisensorThingHandler.java]: reduce to debug

**Comment by kaikreuzer** [comment]
> I feel that we should look for a framework solution for migrating thing-types for non-managed providers.

Such a migration should not be necessary. If the user declared a thing of a more generic type, he will be happy with the more generic features as well. If he wants to use specific features, he should also use a more specific thing type.

**Comment by J-N-K** [comment]
We can leave it as it is then, the "old" thing type will be dynamically upgraded during runtime. This persists if the thing was defined in PaperUI and will happen again if configured in text files. Is it possible to hide the old thing types in PaperUI (so that new things are always created with the new types)?

**Comment by kaikreuzer** [comment]
> Is it possible to hide the old thing types in PaperUI (so that new things are always created with the new types)?

You can add a [listed="false](https://github.com/eclipse/smarthome/blob/7176f8961a4caba501fe38bab52a44c8705011c3/extensions/binding/org.eclipse.smarthome.binding.lirc/ESH-INF/thing/thing-types.xml#L7) to the thing type, which will hide it when manually creating a Thing in Paper UI. Is this what you have in mind?

Btw, we usually do not implement such migration logic and rather declare a breaking change - but as you have done it already, it's ok for me that way; I would just suggest to remove it again after the next release.

**Comment by J-N-K** [comment]
I have been thinking about the migration logic. 

The problem is that PaperUI does not allow to change the thing type without deleting and re-adding. So you would need to reconfigure all things.

For Textual defined things - in this case - you would just have to replace ms-tv and ms-th with ms-tx which is a simple search/replace.

It is clearly out of scope here, but WDYT about a „change thing-type“ Option in PaperUI?

**Comment by ggzengel** [comment]
I use 0.11.0.201812300847 and after adding 20 not existent sensors and than removing them I still have 4 left over:
```
12:47:55.257 [DEBUG] [e.internal.handler.OwBaseThingHandler] - updating thing properties for onewire:temperature:server1:Ow_T_KG_SP_PF_08 failed, no bridge available
12:47:55.257 [DEBUG] [e.internal.handler.OwBaseThingHandler] - updating thing properties for onewire:temperature:server1:Ow_T_KG_SP_PF_09 failed, no bridge available
12:47:55.258 [DEBUG] [e.internal.handler.OwBaseThingHandler] - updating thing properties for onewire:temperature:server1:Ow_T_KG_SP_PF_14 failed, no bridge available
12:47:55.258 [DEBUG] [e.internal.handler.OwBaseThingHandler] - updating thing properties for onewire:temperature:server1:Ow_T_KG_SP_PF_13 failed, no bridge available
```

File: extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/OwBaseBridgeHandler.java [hunk 5]
```diff
@@ -49,35 +52,70 @@ public OwBaseBridgeHandler(Bridge bridge) {
...
             int thingCount = thingList.size();
-            logger.trace("called owRefresher at {}, {} childs", now, thingCount);
-
             Iterator<Thing> childListIterator = thingList.iterator();
+            logger.trace("refreshTask starts at {}, {} childs", now, thingCount);
             while (childListIterator.hasNext() && refreshable) {
                 Thing owThing = childListIterator.next();
 
-                logger.trace("getting handler for {} ({} to go)", owThing.getLabel(), thingCount);
+                logger.trace("refresh: etting handler for {} ({} to go)", owThing.getUID(), thingCount);
```
Comment (ggzengel) [inline extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/OwBaseBridgeHandler.java]: Typo: etting > getting

**Comment by J-N-K** [comment]
@kaikreuzer sorry for adding so many commits after your review, however fixing most of these bugs indepent from the refactoring of the multisensor would have caused lots of work to be done afterwards. I'll not add any further commits except addressinf review commits here now. I have squashed the commits to useful chunks, so that hopefully makes reviewing a bit easier. Thanks in advance.

**Pull request merged**